Egg Move Support

haven1433 opened the pull request:
In the games, there are some moves that can be taught to pokemon only by breeding. For example, Bulbasaur can't learn "Magical Leaf" through a TM, a tutor, or by leveling up. But if you breed a female bulbasaur with a pokemon that knows Magical Leaf, then the baby bulbasaur will have magical leaf. This update adds support for the egg move data within the game.

* The stream of egg moves is automatically located.
* The stream is rendered using the names of pokemon and moves, making the data easy to understand.
* The stream can be edited inline, or via copy/paste, or via the text tool.
* Inline edits include auto-complete.
* Egg move 'limiter' value is automatically updated, allowing you to freely increase and decrease the number of egg moves in the game.
* Find will now find egg move data when you search for a pokemon or move.

hjk321 commented:
How is all of this table formatting metadata stored? .ini files or hardcoded into the program? Looking at the repo, I don't see any data file anywhere, so unless I'm missing something going too much further on adding tables and other features could cause complications in the future. 

And besides, people might want to add or change formatting for their hack, so maybe .ini support should be pretty soon not just for users but for coherent development of the tool itself.

Those are just my two bits when it comes to next steps. Sorry if I rambled a bit. As for this PR I tried it out and the controls are quite a bit more fluid than when I tried it last during the usability test. Tables are somewhat improved too. As for as ctrl+click or right-clicking to open the egg tool I had no idea that was a thing except by reading the commits so idk if there's a way to make this more intuitive. Probably just by adding ctrl+click actions for a lot of other stuff too so it's somewhat consistent.

haven1433 commented:
hjk,

Don't worry about rambling! The only way I can understand what other people find confusing, is if they tell me :)

Regarding metadata:
* I have an auto-search algorithm (See [AutoSearchModel](https://github.com/haven1433/HexManiacAdvance/blob/master/src/HexManiac.Core/Models/AutoSearchModel.cs) that's in charge of dynamically finding the data. The locations aren't hard-coded, but the expected format is.
* Each file the user edits is given a matching TOML file (See [StoredMetadata](https://github.com/haven1433/HexManiacAdvance/blob/master/src/HexManiac.Core/Models/StoredMetadata.cs)) that is human readable and hand-editable. So even if the auto-search fails or is wrong, the user *can* still edit it.
This was a good question! I've updated the Wiki to explain this more than I had before. See [Metadata](https://github.com/haven1433/HexManiacAdvance/wiki/Metadata).

Regarding Ctrl+Click / Right-Click:
* Right-Click is fairly well established as a way to open a context menu to get extra options. I've added Right-Click menu options to anchors, pointers, text, tables, and now egg data. I plan to continue adding Right-Click support for new features, so hopefully my users will just come to expect right-click as a way to get more options.
* Ctrl+Click does have discover-ability problems, but it's also not required. Likewise, shortcuts like Ctrl+C, Ctrl+X, Ctrl+V, Ctrl+G, Ctrl+F, Ctrl+Z, and Ctrl+Y are all fairly common shortcuts, but fairly difficult to intuit. As such, whenever you see a Right-Click menu item (or just an item in the menu) with a shortcut associated with it, I display the shortcut next to it to help make it more discoverable. However, if this continues to be an issue, I'll try to think of other strategies to help.
